pow/src/main/java/tech/pegasys/teku/pow/FallbackAwareEth1Provider.java

```diff
@@ -0,0 +1,146 @@
+              return asyncRunner
+                  .getDelayedFuture(Constants.ETH1_INDIVIDUAL_BLOCK_RETRY_TIMEOUT)
+                  .thenCompose(__ -> getGuaranteedEth1Block(blockHash));
+            });
+  }
+
+  @Override
+  public SafeFuture<Block> getGuaranteedEth1Block(final UInt64 blockNumber) {
+    return run(eth1Provider -> eth1Provider.getEth1Block(blockNumber))
+        .thenApply(Optional::get)
+        .exceptionallyCompose(
+            (err) -> {
+              LOG.debug("Retrying Eth1 request for block: {}", blockNumber, err);
+              return asyncRunner
+                  .getDelayedFuture(Constants.ETH1_INDIVIDUAL_BLOCK_RETRY_TIMEOUT)
+                  .thenCompose(__ -> getGuaranteedEth1Block(blockNumber));
+            });
+  }
+
+  @Override
+  public SafeFuture<Optional<Block>> getEth1Block(final String blockHash) {
+    return run(eth1Provider -> eth1Provider.getEth1Block(blockHash));
+  }
+
+  @Override
+  public SafeFuture<Optional<Block>> getEth1BlockWithRetry(
+      final String blockHash, final Duration retryDelay, final int maxRetries) {
+
+    return asyncRunner.runWithRetry(
+        () -> run(eth1Provider -> eth1Provider.getEth1Block(blockHash)), retryDelay, maxRetries);
+  }
+
+  @Override
+  public SafeFuture<Block> getLatestEth1Block() {
+    return run(Eth1Provider::getLatestEth1Block);
+  }
+
+  @Override
+  public SafeFuture<EthCall> ethCall(
+      final String from, final String to, final String data, final UInt64 blockNumber) {
+    return run(eth1Provider -> eth1Provider.ethCall(from, to, data, blockNumber));
+  }
+
+  @Override
+  public SafeFuture<BigInteger> getChainId() {
+    return run(Eth1Provider::getChainId);
+  }
+
+  @Override
+  public SafeFuture<Boolean> ethSyncing() {
+    return run(Eth1Provider::ethSyncing);
+  }
+
+  @Override
+  public SafeFuture<List<EthLog.LogResult<?>>> ethGetLogs(EthFilter ethFilter) {
+    return run(eth1Provider -> eth1Provider.ethGetLogs(ethFilter));
+  }
+
+  private <T> SafeFuture<T> run(final Function<Eth1Provider, SafeFuture<T>> task) {
+    return run(task, eth1ProviderSelector.getProviders().iterator(), new Eth1RequestException());
+  }
+
+  private <T> SafeFuture<T> run(
+      final Function<Eth1Provider, SafeFuture<T>> task,
+      final Iterator<Eth1Provider> providers,
+      Eth1RequestException exceptionsContainer) {
+    return SafeFuture.of(
+        () ->
+            task.apply(providers.next())
+                .exceptionallyCompose(
+                    err -> {
+                      exceptionsContainer.addSuppressed(err);
+                      if (providers.hasNext()) {
+                        LOG.warn("Retrying with next eth1 endpoint", err);
+                        return run(task, providers, exceptionsContainer);
+                      } else {
+                        LOG.error("All available eth1 endpoints failed", exceptionsContainer);
+                        return SafeFuture.failedFuture(exceptionsContainer);
```

Comment:
In testing locally I was reminded that we deliberately don't print every error from the eth1 node above debug level because there are so many requests made and retries that they wind up swamping the logs. Otherwise this works great so I'm going to change these to debug level, pull in the latest changes and merge this.

I've logged https://github.com/ConsenSys/teku/issues/3855 to improve the logging around eth1 - it's not terrible now but we can do better and some of it may be worth at least bearing in mind when doing the status checks on endpoints.